File: Model/punto.cpp
#include "punto.h"
#include <math.h>

Punto::Punto(double a, double b) : x(a), y(b) {}

double Punto::getX() const {
    return x;
}

double Punto::getY() const {
    return y;
}

Punto Punto::getPunto() const {
    return Punto(x,y);
}

void Punto::setPunto(const Punto& p) {
    x = p.getX();
    y = p.getY();
}

void Punto::setX(double i) {
    x = i;
}

void Punto::setY(double i) {
    y = i;
}


double Punto::distanzaDuePunti(const Punto& a, const Punto& b) {
    double v = pow(b.getX() - a.getX(), 2);
    double z = pow(b.getY() - a.getY(), 2);
    double dist = sqrt(v+z);
    return dist;

}

Punto* Punto::clone() const {
    return new Punto(*this);
}

File: SuperQtKalkJAVA/src/Punto.java
import static java.lang.Math.pow;
import static java.lang.Math.sqrt;

public class Punto {
    private double x;
    private double y;

    public Punto(double i, double j) {
        x = i;
        y = j;
    }

    public double getX() {
        return x;
    }
    public double getY() {
        return y;
    }

    public void setPunto(Punto p) {
        x = p.getX();
        y = p.getY();
    }

    public void setX(double i) {
        x = i;
    }

    public void setY(double i) {
        y = i;
    }

    public double distanza(Punto p) {
        double a = pow(p.getX()-x,2);
        double b = pow(p.getY()-y,2);
        return sqrt(a+b);
    }

    static double distanzaDuePunti(Punto a, Punto b) {
        double v = pow(b.getX() - a.getX(), 2);
        double z = pow(b.getY() - a.getY(), 2);
        double dist = sqrt(v+z);
        return dist;
    }
}
